test-data/run.test

```diff
@@ -2523,78 +2528,29 @@ class A(object):
         yield self.x
 
 [file driver.py]
-from native import yield_three_times, yield_twice_and_return, yield_while_loop, yield_for_loop, yield_with_except, complex_yield, A
-
-generator = yield_three_times()
-assert next(generator) == 1
-assert next(generator) == 2
-assert next(generator) == 3
-try:
-    next(generator)
-except StopIteration as e:
-    assert e.value is None
+from typing import Generator, Tuple, TypeVar, Sequence, Union
+from native import yield_three_times, yield_twice_and_return, yield_while_loop, yield_for_loop, yield_with_except, complex_yield, yield_with_default, A
 
-generator = yield_twice_and_return()
-assert next(generator) == 1
-assert next(generator) == 2
-try:
-    next(generator)
-except StopIteration as e:
-    assert e.value == 4
-try:
-    next(generator)
-except StopIteration as e:
-    assert e.value is None
-
-generator = yield_while_loop()
-assert next(generator) == 0
-assert next(generator) == 1
-assert next(generator) == 2
-try:
-    next(generator)
-except StopIteration as e:
-    assert e.value == 3
-try:
-    next(generator)
-except StopIteration as e:
-    assert e.value is None
+T = TypeVar('T')
+U = TypeVar('U')
 
-generator = yield_for_loop()
-for i in range(4):
-    assert next(generator) == 0
-    assert next(generator) == 1
-    assert next(generator) == 2
-try:
-    next(generator)
-except StopIteration as e:
-    assert e.value is None
+def run_generator(gen: Generator[T, None, U]) -> Tuple[Sequence[T], Union[U, str]]:
```

Comment:
Don't need the `Union` in the return type; that was for the version that could catch other exceptions than `StopIteration`.

Reply:
Fixed.